src/core/index.js

```diff
@@ -3,17 +3,27 @@ const getExternals = require('./externals')
 const getManifest = require('./manifest')
 const staticAssets = require('./static')
 const getCompiler = require('./compiler')
+const mergeConfig = require('./merge')
 
 module.exports = async (env, config, bar) => {
+  // Load only supported syntaxes to reduce bundle size
   await loadSyntax(config)
-  await getExternals(config)
+
+  // Fetch any external docs sources
+  const externals = await getExternals(config)
+
+  // Merge current and extenal configs
+  const mergedConfig = mergeConfig(config, externals)
+
+  // Load static assets like images, scripts, css, etc.
   await staticAssets(config, env === 'development')
 
-  const manifest = await getManifest(env, config)
+  // Build the documentation manifest
+  const manifest = await getManifest(env, mergedConfig)
 
   // this gets passed to the theme app
   const props = {
-    config,
+    mergedConfig,
```

Comment:
Other files depend on `props.config` so the key name should stay the same here. Otherwise it will break a lot of stuff.